### PR TITLE
Fs mandatory tasks 07

### PR DESCRIPTION
**Pull request by fsher**
Review 06 first please.

### CODE REVIEW

File: 07-regexp-mandatory-tasks/src/com/fsher/exercise2/Exercise2.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.fsher.exercise2;
+
+public class Exercise2 {
+    private static final String REGEXP_STRING = "     Removing     Extra Spaces    Between Words in    the text      ";
+
+    public static void main(String[] args) {
+        System.out.println(
+                REGEXP_STRING
+                        .replaceAll("(^[\\s]+)|([\\s]+$)", "")
+                        .replaceAll("(\\s{2,})", " ")
```
Comment (ssheludchenko) [inline 07-regexp-mandatory-tasks/src/com/fsher/exercise2/Exercise2.java:10]: Combine in a single regex?
`(^\\s+)|(\\s+$)|(\\s{2,})`

File: 07-regexp-mandatory-tasks/src/com/fsher/exercise2/Exercise2.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.fsher.exercise2;
+
+public class Exercise2 {
+    private static final String REGEXP_STRING = "     Removing     Extra Spaces    Between Words in    the text      ";
+
+    public static void main(String[] args) {
+        System.out.println(
+                REGEXP_STRING
+                        .replaceAll("(^[\\s]+)|([\\s]+$)", "")
```
Comment (ssheludchenko) [inline 07-regexp-mandatory-tasks/src/com/fsher/exercise2/Exercise2.java]: `[]` seems redundant

File: 07-regexp-mandatory-tasks/src/com/fsher/exercise3/Exercise3.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.util.Arrays;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class Exercise3 {
+    private static final String REGEX_STRING = "password wwwword word Helllloooooooo";
+
+    public static void main(String[] args) {
+        final Predicate<String> match = str -> str.matches("[\\S]*(\\S)\\1[\\S]*");
```
Comment (ssheludchenko) [inline 07-regexp-mandatory-tasks/src/com/fsher/exercise3/Exercise3.java]: `[]` seems redundant

**Review by ssheludchenko — APPROVED** [review]

File: 07-regexp-mandatory-tasks/src/com/fsher/exercise2/Exercise2.java (hunk 1 above)
Comment (fsher) [reply 07-regexp-mandatory-tasks/src/com/fsher/exercise2/Exercise2.java:10]: @ssheludchenko We need to perform different operations to these regexes

File: 07-regexp-mandatory-tasks/src/com/fsher/exercise2/Exercise2.java (hunk 2 above)
Comment (fsher) [reply 07-regexp-mandatory-tasks/src/com/fsher/exercise2/Exercise2.java]: Thanks, done.

File: 07-regexp-mandatory-tasks/src/com/fsher/exercise3/Exercise3.java (hunk 3 above)
Comment (fsher) [reply 07-regexp-mandatory-tasks/src/com/fsher/exercise3/Exercise3.java]: Thanks, done!

**Pull request merged**